### DISCUSSION TITLE
Equivalent of inject-st?

### DISCUSSION THREAD

**Discussion by zhaoyig** [koka-lang/koka#590] [Q&A]
I'm trying to implement `interleaved` from this technical report by Daan: https://www.microsoft.com/en-us/research/wp-content/uploads/2017/05/asynceffects-msr-tr-2017-21.pdf

> The function `inject-st` is used to injects the `st⟨h⟩` effect into a function effect `e`:
`inject-st : (() → e a) → total (() → ⟨st⟨h⟩|e⟩ a)`

Anyone know how can I achieve this in the latest release of Koka?

**Comment by TimWhiting** [comment]
See here for the (in-progress) implementation of async for the C backend. 

https://github.com/koka-lang/koka/pull/384/files#diff-b76d9348d716357950f9873ffd1699bf605fd1c066f66715e1b5d17527a62023

I think you are looking for `mask behind<..>{}`
It's talked about a bit more in the book under the overriding handlers section: https://koka-lang.github.io/koka/doc/book.html#sec-overriding-handlers
